src/js/hooks/productSupplier/form/usePreferenceTypeVariationsColumns.jsx

```diff
@@ -28,89 +35,148 @@ const usePreferenceTypeVariationsColumns = ({ errors, control, remove }) => {
     dispatch(fetchBuyers());
   }, []);
 
+  const isFieldValid = (index, property) => {
+    const hasFieldErrors = errors?.[index]?.[property]?.message
+                   || errors?.[index]?.[property]?.id?.message;
+    const isRowDirty = _.some(Object.values(updatedRows?.[index] || {}));
+
+    if (!isRowDirty) {
+      return true;
+    }
+
+    return !hasFieldErrors;
+  };
+
+  const getCustomSelectErrorPlaceholder = ({
+    id,
+    defaultMessage,
+    displayIcon,
+  }) => (
+    <div className="custom-select-error-placeholder">
+      {displayIcon && <RiErrorWarningLine />}
+      <span>
+        <Translate
+          id={id}
+          defaultMessage={defaultMessage}
+        />
+      </span>
+    </div>
+  );
+
   const columns = useMemo(() => [
     {
       Header: translate('react.productSupplier.table.siteName.label', 'Site Name'),
       sortable: false,
       accessor: 'destinationParty',
-      minWidth: 300,
+      minWidth: 276,
       Cell: (row) => (
         <Controller
           key={row.original.id}
           name={`productSupplierPreferences.${row.index}.destinationParty`}
           control={control}
-          render={({ field }) => (
-            <SelectField
-              options={buyers}
-              errorMessage={
-                errors?.[row.index]?.destinationParty?.message
-              }
-              {...field}
-            />
-          )}
+          render={({ field }) => {
+            const hasErrors = !isFieldValid(row.index, 'destinationParty');
+            return (
+              <SelectField
+                options={buyers}
+                errorMessage={hasErrors}
```

Comment:
this is why I said something didn't feel right to me - imho it feels like a workaround. `errorMessage` is supposed to receive a string message and in this case we pass a boolean.
I think if we need such a case, we should think about making the `errorMessage` as optional prop and add a new one `hasErrors` that would handle the logic you need. 
I think it would be better than mixing the expected type of the `errorMessage` and adding `displayErrorMessage` prop.

Reply:
Basically, I added a `displayErrorMessage` prop to avoid having this `hasErrors` :laughing: 
The idea of this is to avoid passing `hasErrors` and `errorMessage` each time. For now, we can only pass `errorMessage` and this prop is an indicator of whether we have to display an error. The `displayErrorsMessage` is an additional prop turning on/off the captions under the input field. For me, this is a little bit easier to use than the solution you proposed. I am interested in what other developers think about that, so If they have similar feelings as you, I can change it.

cc @awalkowiak @drodzewicz

Reply:
on one hand you say:

> `errorMessage` and this prop is an **indicator of whether we have to display an error**.

so if `errorMessage` is the **indicator**, then what the `displayErrorMessage` is? Imho the `errorMessage` should NOT be the indicator, because the name says error message, so we expect it to be a string, not a boolean/an indicator. 

If we wanted somewhat a compromise, I'd suggest either (I think not recommended since you would have to change it in every place that has been done) changing from `errorMessage` to `error` object that would receive `message` and `showMessage` properties or......

I'll let the others speak, but I strongly feel like we should not mix the destination of the `errorMessage` to use it once as a string and once as somewhat an indicator and pass a boolean.

Reply:
Eh, the `errorMessage` is a string, if it has content then the error is displayed, if it doesn't the the content is not visible (this is what I meant by indicator here).
The `displayErrorMessage` is just a prop for showing the caption under the input. If we don't pass it, then the error is visible by default.

Reply:
Yeah, it should be a string, but you actually pass there `hasErrors` which is a boolean.
I still don't feel convinced about the approach on that. @drodzewicz @awalkowiak what do you think?

Reply:
@alannadolny you can do a workaround to Kacpers requirements :grin:  
```jsx
const errorMessage = isFieldValid(row.index, 'destinationParty') ? null : 'validation error';
return (
   <SelectField
     options={buyers}
     errorMessage={errorMessage}
     ...
```

But if we want to be strict with ourselves I think we would need to add an additional `isValid` prop which would trigger some error classes and such.
Basically, `errorMessage` prop should only be responsible for the error message and `isValid` for everything else like red border classes etc..